### PR TITLE
ci: add slither job

### PR DESCRIPTION
**Pull request by scorpion9979**
Addresses https://github.com/sablier-labs/v2-core/issues/418.
Recreated from https://github.com/sablier-labs/v2-core/pull/431.

### CODE REVIEW

**Comment by scorpion9979** [comment]
@andreivladbrg I've updated the GitHub action, as it was failing due to attempting to call `forge` from the Slither job's Docker container.

Context:
- [job run](https://github.com/sablier-labs/v2-core/actions/runs/6590133819/job/17912620324#step:4:215)
- [comment on a slither-action issue](https://github.com/crytic/slither-action/issues/44#issuecomment-1644201981)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -284,4 +284,47 @@ jobs:
...
+            out
+            out-optimized
+
+      - name: "Run Slither analysis"
+        uses: "crytic/slither-action@v0.3.0"
+        id: "slither"
+        with:
+          fail-on: "none"
+          sarif: "results.sarif"
+          solc-version: "0.8.19"
```
Comment (andreivladbrg) [inline .github/workflows/ci.yml:318]: we are using solc `0.8.21` now

**Comment by andreivladbrg** [comment]
It looks like the slither job keeps on failing, I had a look and I couldn't figure it out why.

@scorpion9979 can you help us?

**Comment by scorpion9979** [comment]
@andreivladbrg It seems like the GitHub action is still attempting to call `forge` from inside the job's Docker container. I found another possible workaround at https://github.com/crytic/slither-action/issues/44#issuecomment-1338183656, which I just pushed a commit for. It might take a few trials to get this right.

**Comment by andreivladbrg** [comment]
@scorpion9979  It looks like the slither job has [passed](https://github.com/sablier-labs/v2-core/actions/runs/6622915809/job/17995619673?pr=714).Am I missing something?
 The fork tests are failing because you probably don't have set up `RPC_URL_MAINNET ` in the github secrets of your [repo](https://github.com/scorpion9979/sablier-v2-core/).

**Comment by scorpion9979** [comment]
@andreivladbrg Thanks! I just re-ran all CI jobs after setting up the GitHub secrets in my forked repo and all are now passing. I just double-checked to make sure the Slither job is as clean as possible, and I think this PR is now ready to be reviewed and merged.

**Comment by andreivladbrg** [comment]
The fork tests have failed again. I've seen on your fork that they worked, and I also tested them locally and they worked.
I'm not sure what the issue is now

**Review by andreivladbrg — APPROVED** [review]
Looks good.

Good job @scorpion9979

**Comment by PaulRBerg** [comment]
Thanks again for your contribution, @scorpion9979!

@andreivladbrg did you look over the Slither report to check whether it reported something that we should address

**Comment by andreivladbrg** [comment]
> did you look over the Slither report to check whether it reported something that we should address?

Yeah, I did and found nothing that we should address. You can see the report [here](https://github.com/sablier-labs/v2-core/security/code-scanning?query=pr%3A714+is%3Aopen).

**Comment by PaulRBerg** [comment]
Great.

> You can see the report [here](https://github.com/sablier-labs/v2-core/security/code-scanning?query=pr%3A714+is%3Aopen).

Could you please mark all items in the code scan report as "False positive", "Used in tests", or "Won't fix"?

SCR-20231027-omso

**Pull request merged**